scheduler/feasible.go

```diff
@@ -96,6 +96,69 @@ func NewRandomIterator(ctx Context, nodes []*structs.Node) *StaticIterator {
 	return NewStaticIterator(ctx, nodes)
 }
 
+// HostVolumeChecker is a FeasibilityChecker which returns whether a node has
+// the host volumes necessary to schedule a task group.
```

Comment:
We do allow this for host volumes - was planning on documenting this as part of the feature documentation tho. There are plenty of cases where this could be useful for ML workloads and similar.